src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn4.0.cs

```diff
@@ -24,70 +25,67 @@ public sealed partial class JsonSourceGenerator : IIncrementalGenerator
         public void Initialize(IncrementalGeneratorInitializationContext context)
         {
 #if LAUNCH_DEBUGGER
-            Diagnostics.Debugger.Launch();
+            System.Diagnostics.Debugger.Launch();
 #endif
-            IncrementalValuesProvider<ClassDeclarationSyntax> classDeclarations = context.SyntaxProvider
+            IncrementalValueProvider<KnownTypeSymbols> knownTypeSymbols = context.CompilationProvider
+                .Select((compilation, _) => new KnownTypeSymbols(compilation));
```

Comment:
I have to ask, is calculating the KnownTypeSymbols,~50 individual calls to GetBestTypeByMetadataName, each each and every time the srcgen is invoked better than just doing it in the Parser, if the appropriate knownTypeSymbol is required, better (or caching it there)?

How many individual Json Contexts do you expect per solution? I guess 50 at most (for even large solutions I guess, even one of my larger solutions talking to a db with json storage it's only around 20)

From my own srcgen experience, I removed the calculation of a dedicated KnownTypeSymbols step from my srcgens and from a (non-measured) impression VS was snappier after it, my KnownTypeSymbols only contained like 20 or so symbols and 4-5 were used all the time, so I just moved the into the Parser.